ConfigSpace/c_util.pyx

```diff
@@ -315,6 +322,16 @@ cpdef np.ndarray change_hp_value(
                     active = False
                     break
 
+            if active:
+                hps_to_be_activate.add(current_idx)
+                if current_value == current_value:
```

Comment:
This seems odd? `current_value == current_value`, is this not always true?

Reply:
Not really, if current_value is deactivated, then this equation does not hold (Similar code can be found under line 335 and 344)